### ISSUE TITLE
Auto-trigger only works on first segment

### ISSUE THREAD

**Issue by joefowler** [usnistgov/dastard#16]
There's a problem in the auto-triggering. Need to investigate and fix.

**Comment by joefowler** [comment]
Aha! All data sources are supposed to generate `DataSegment` objects that include two key indicators of time passing: a `firstFramenum` giving the monotonically increasing counter of all frames ever, and a `firstTime` to indicate the real-world time corresponding to the first frame in the segment.

```go
type DataSegment struct {
	rawData         []RawType
	framesPerSample int // Normally 1, but can be larger if decimated
	firstFramenum   FrameIndex
	firstTime       time.Time
	framePeriod     time.Duration
}
```

**But if these aren't filled properly it creates this issue.** I'll generate a test for it, but at the moment, none of the working types that implement `DataSource` are updating either the counter or the timer with each new segment. This includes `TriangleSource`, `SimPulseSource`, and `LanceroSource`. 

This is why auto-trigger was failing: because data was appearing not to proceed into the future!

**Comment by joefowler** [comment]
Fixed in 7abe3a5
